### DISCUSSION TITLE
Convertation to *.vsgt or *.vsgb from 3D Editors

### DISCUSSION THREAD

**Discussion by maisvendoo** [vsg-dev/VulkanSceneGraph#951] [Q&A] [answered]
Hi! I have one question.
How I can to convert 3D-model to VSG native format from 3D Editors, such as 3d max or Blender?

**Comment by robertosfield** [comment, marked as answer]
The [vsgXchange](https://github.com/vsg-dev/vsgXchange) library uses [assimp](https://github.com/assimp/assimp) to read 3rd party model formats, once loaded then can be written to .vsgt and .vsgb.  Usage is in the form:

~~~ cpp
// set up the options for reading/writing, such as what ReaderWriter's to use
auto options = vsg::Options::create();
options->add(vsgXchange::all::create());

if (auto model = vsg::read("mymodel.gltf", options))
{
    vsg::write(model, "mymodel.vsgb", options);
}
~~~

Or you can use the vsgconv utility that comes with vsgXchange:

~~~ sh
vsgconv mymodel.gltf mymodel.vsgb
~~~

To list all the image and 3d model formats supported use:
~~~ sh
vsgconv --features
~~~

The [vsgXchange/README.md](https://github.com/vsg-dev/vsgXchange#file-formats-supported-by-all-built-in-readerwriters) also contains this list of supported formats.

**Comment by maisvendoo** [comment]
Thanks for youre reply!
I read Tutorial on youre site about support assimp and vsgconv. I thought about idea of convertation to VSG from other formats. But I hoped that exist direct mothod of convertion, such as convert from 3d max and Blender to OpenSceneGraph formats. 
Ok, I think, I can solved this task for VSG by myself.

**Reply by robertosfield** [reply]
"3d max and Blender to OpenSceneGraph formats" ?  Do you mean OpenSceneGraph or VulkanSceneGraph?

Modelling tools export a range of different formats and assimp supports a range of formats so it's a case of finding out what combination of supported formats works best for you.  Potentially one can write plugins for Modellers to export to native VSG formats, but this isn't something I'm personally planning on tackling.

Converting to native VSG formats may not be necessary, it all depends on the requirements of your application.
